### DISCUSSION TITLE
cocotb compatible simulator versions

### DISCUSSION THREAD

**Discussion by sammy17** [cocotb/cocotb#2673] [Q&A]
Hi, I could not find the versions of the simulators that are compatible with cocotb. 
I got to know after a bit of searching that only the Verilator 4.106 works correctly with cocotb. 

Is there a similar version restriction on VCS?

I tried my design with vcs/Q-2020.03-SP2-2 and seems like it hanged, I wanted to check whether there are any version requirements before further debugging. 

**NOTE:** This design runs fine with verilator 4.106. 

**My situation:** My requirement is to get RTL code coverage with either VCS or Verilator. With verilator I ran in to issues which I have filed as https://github.com/cocotb/cocotb/issues/2573 in cocotb and https://github.com/verilator/verilator/issues/3039. Since there was no easy way of fixing this with verilator, I am looking into simulation with VCS. 

Any comments/directions on how to proceed are appreciated!

**Comment by cmarqu** [comment]
Simulation with VCS seems to be flaky across multiple versions, see https://github.com/cocotb/cocotb/issues/1261
In case you get the same crash, there is a workaround mentioned that you could try.

**Reply by sammy17** [reply]
Thanks for the response, I had a crash and did the workaround mentioned in https://github.com/cocotb/cocotb/issues/1163 which fixed it. I anyway did the workaround that you sent too. 
Currently, I can see the simulation starting and clock running but I don't see the desired outcome. Simulation reaches the maximum number of clock cycles. Please let me know if there are any other known issues related to this. 

I see below warnings during the VCS compilation, is this expected?
```
Warning-[VPI-CT-NS] VPI function is not supported
  The VPI function 'vpi_register_cb' is not supported in the compiler.
  Please fix the code being referred to by the '-load' compile option.

Warning-[VPI-CT-NS] VPI function is not supported
  The VPI function 'vpi_chk_error' is not supported in the compiler.
  Please fix the code being referred to by the '-load' compile option.
```
